src/system_reader.rs

```diff
@@ -265,7 +286,66 @@ impl<R: RawReader> SystemReader<R> {
                 }
             }
             ProcessingLstImports => {
-                return decoding_error("Importing shared symbol tables is not yet supported.");
+                // We're in the `imports` list processing a shared symbol table import.
+                if let (IonType::Struct, false) = (ion_type, is_null) {
+                    if self.lst.current_import_name.is_some()
+                        && self.lst.current_import_version.is_some()
+                    {
```

Comment:
There are a lot of conditionals to track in this segment, and I think it's because we're "squishing" all of the different reader positions in `imports` processing into a single state: `ProcessingLstImports`. We should add some new states to break things up.

* When the reader encounters an LST, its state is `AtLstStart`.
* If the reader steps in and encounters the field name `imports`, its state is `AtLstImports`.
* If the reader steps into the `imports` list, its state is `ProcessingLstImports`.
* If the reader steps in and encounters ______, its state is `______`.

The new states might be:
* `"name"` -> `AtImportName`
* `"version"` -> `AtImportVersion`
* `"max_id"` -> `AtImportMaxId`

Being on one of these states would tell you the reader's depth, the field name, (implicitly) the path it took to get there, what stream item to expect next, etc. Far fewer `if`s should be needed, though you'll end up with more arms in the `match` statements.

When `step_out()` is called while the reader is in one of the new states, you can `finish_reading_current_level()` and then reset all of the imports-related `self.lst` fields.